Add PHP7.1 to test matrix.

markstory opened the pull request:
Get started on PHP7.1 support for 2.x

markstory committed:
Add PHP7.1 to test matrix.

ravage84 commented on .travis.yml:
```diff
@@ -6,6 +6,7 @@ php:
   - 5.5
   - 5.6
   - 7.0
+  - 7.1
```
Could we put it into a ``allow_failures`` section for the time being?

markstory replied:
I was hoping to get the tests passing in this pull request as well unless it is a huge diff.

markstory committed:
Fix errors in php7.1

* The constructor of errors has changed in PHP 7.1
* mcrypt is no longer available in PHP 7.1 by default.

ravage84 commented on lib/Cake/Test/Case/Controller/Component/Auth/ControllerAuthorizeTest.php:
```diff
@@ -47,15 +47,14 @@ public function setUp() {
+			$this->fail('No exception thrown');
 		} catch (Throwable $t) {
-			throw new PHPUnit_Framework_Error($t);
+			$this->assertTrue(true, 'Exception was raised');
```
Clever 😼

markstory replied:
Not ideal, but given the PHPUnit version and PHP versions we are supporting it works.

markstory committed:
Fix type error tests to work in PHP5 & PHP7.1

Catch the TypeErrors that are raised and make the match the PHP5
behavior of a converted error.

markstory committed:
Fix PHPCS errors.
